doc/api/module.md

```diff
@@ -217,6 +217,39 @@ added: v22.8.0
 * Returns: {string|undefined} Path to the [module compile cache][] directory if it is enabled,
   or `undefined` otherwise.
 
+### `module.getPackageJSON(startPath[, everything])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active Development
+
+* `startPath` {string} Where to start looking
+* `everything` {boolean} Whether to return the full contents of the found package.json
```

Comment:
I'm not sure it is a good idea to introduce yet-another way of loading a JSON file, wouldn't it be better to return the path and let the user deal with reading/parsing the file?

Reply:
I considered that (there's another PR I opened and closed that exposed `findPackageJSON`, which did exactly that). We're already reading the pjson and caching it, so IMO that would force a de-op as well as force the user to manually do what we're already doing. So, why?

Reply:
I think returning the parsed data means every time we add consider parsing another field in the package.json, it needs to be surfaced into this API, even though that field may not even be useful for tools, but we'll end up paying the serialization/deserialization cost even though internally only selected places need selected fields. IMO even just returning the string would be better than this. Users need this for the lookup algorithm, they tend to have other fields to parse anyway.

Reply:
Sorry, @joyeecheung I'm not sure I follow in your message. I think you've misunderstood the behaviour here: internally, we do not use the full version—we do use `getNearestParentPackageJSON` but do not set `everything` to `true`. When `everything` is not `true`, only the select fields we use internally are parsed. That has not changed in this PR.

Reply:
> May I recommend a different naming and API surface?
> 
> `module.findPackageJson(path, { includeContents: true })`

@anonrig if you recall, when we were working on this a few days ago, that did not make sense due to how the C++ worked, which is what caused me to abandon the `findPackageJSON` + `getPackageJSON` route in the first place.

Regarding "find" + `includeContents`, this seems counter-intuitive to me: I expect something called "find" to only locate, not retrieve. I could perhaps see the inverse

`module.getPackageJSON(path, { includeContents: false })`

I do not like either option though because the shape of the return is changed significantly in a foot-gun way:

```
getPackageJSON(path, false) // '…/package.json'

getPackageJSON(path, true)  // { data: {…}, path: '…/package.json' }
```
(I simplified the 2nd arg for brevity, not necessarily to say we shouldn't use an object)

I suppose it could always return an object for consistency, but that seems like a clumsy compromise

```
getPackageJSON(path, false) // { path: '…/package.json' }

getPackageJSON(path, true)  // { data: {…}, path: '…/package.json' }
```

If we were to do this, I think exposing 2 different utils (`findPackageJSON` + `getPackageJSON`) would be better.